app/controllers/signup/base_controller.rb

```diff
@@ -16,7 +18,7 @@ def is_membership_enabled?
     private
 
     def load_member
-      if session[:member_id] && session[:timeout] && session[:timeout] > Time.current - 15.minutes
```

Comment:
This logic was really hard for me to wrap my head around. Now, `session[:timeout]` contains a time, after which the current session is considered invalid.